gtkplus build failure on Ubuntu LTS

On a clean `Ubuntu 16.04.3 LTS` (built via [this Dockerfile](https://gist.github.com/nazavode/50c6336f73fca09cf0c4b0997747df08)) the `configure` phase of `gtkplus` fails complaining about missing `cairo-xlib`. When `gtkplus` is asked for `+X` variant, everything works as expected as `cairo-xlib` is provided by `cairo+X`. I tried to configure the source manually using `./configure --without-x` but it keeps complaining: `cairo-xlib` looks like a mandatory dependency.

### Expected Result

To have the `gtkplus` package built and installed.

### Actual Result

```
==> Error: ProcessError: Command exited with status 1:
    '/home/spack/spack/var/spack/stage/gtkplus-2.24.31-hbqdviwkldktib7stpuem5mapww527hb/gtk+-2.24.31/configure' '--prefix=/home/spack/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/gtkplus-2.24.31-hbqdviwkldktib7stpuem5mapww527hb'

1 error found in build log:
     [ ... ]
     106   checking for library containing strerror... none required
     107   checking whether make sets $(MAKE)... (cached) yes
     108   checking for pkg-config... /home/spack/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/pkgconf-1.4.0-c5lf2hilcinnqpf3nojkif3eauay2gko/bin/pkg-config
     109   checking pkg-config is at least version 0.9.0... yes
     110   checking for BASE_DEPENDENCIES... yes
     111   checking for CAIRO_BACKEND... no
  >> 112   configure: error: Package requirements (cairo-xlib >= 1.6) were not met:
     113
     114   Package 'cairo-xlib', required by 'virtual:world', not found
     115
     116   Consider adjusting the PKG_CONFIG_PATH environment variable if you
     117   installed software in a non-standard prefix.
     118
```

### Steps to reproduce the issue

```console
$ spack install gtkplus
```

### Information on your system

The system used to reproduce this issue has been built via [this Dockerfile](https://gist.github.com/nazavode/50c6336f73fca09cf0c4b0997747df08). It is an `Ubuntu 16.04.3 LTS Server` (no GUI components) running `gcc@5.4.0` and `python@2.7.12`.

--- a/var/spack/repos/builtin/packages/dia/package.py
+++ b/var/spack/repos/builtin/packages/dia/package.py
@@ -33,7 +33,7 @@ class Dia(Package):
     version('0.97.3',    '0e744a0f6a6c4cb6a089e4d955392c3c')
 
     depends_on('intltool', type='build')
-    depends_on('gtkplus@2.6.0:+X')
+    depends_on('gtkplus@2.6.0:')
     depends_on('libxslt')
     depends_on('python')
     depends_on('swig')

--- a/var/spack/repos/builtin/packages/emacs/package.py
+++ b/var/spack/repos/builtin/packages/emacs/package.py
@@ -57,7 +57,7 @@ class Emacs(AutotoolsPackage):
     depends_on('giflib', when='+X')
     depends_on('libx11', when='+X')
     depends_on('libxaw', when='+X toolkit=athena')
-    depends_on('gtkplus+X', when='+X toolkit=gtk')
+    depends_on('gtkplus', when='+X toolkit=gtk')
     depends_on('gnutls', when='+tls')
     depends_on('libxpm ^gettext+libunistring', when='+tls')
     depends_on('ncurses+termlib', when='+tls')

--- a/var/spack/repos/builtin/packages/gtkplus/package.py
+++ b/var/spack/repos/builtin/packages/gtkplus/package.py
@@ -34,18 +34,16 @@ class Gtkplus(AutotoolsPackage):
     version('2.24.31', '68c1922732c7efc08df4656a5366dcc3afdc8791513400dac276009b40954658')
     version('2.24.25', '38af1020cb8ff3d10dda2c8807f11e92af9d2fa4045de61c62eedb7fbc7ea5b3')
 
-    variant('X', default=False, description="Enable an X toolkit")
-
     depends_on('pkgconfig', type='build')
 
-    depends_on("atk")
-    depends_on("gdk-pixbuf")
-    depends_on("glib")
-    depends_on("pango")
-    depends_on("pango~X", when='~X')
-    depends_on("pango+X", when='+X')
-    depends_on('gobject-introspection', when='+X')
+    depends_on('atk')
+    depends_on('gdk-pixbuf')
+    depends_on('glib')
     depends_on('shared-mime-info')
+    # Hardcode X11 support (former +X variant),
+    # see #6940 for rationale:
+    depends_on('pango+X')
+    depends_on('gobject-introspection')
 
     patch('no-demos.patch')
 

--- a/var/spack/repos/builtin/packages/icedtea/package.py
+++ b/var/spack/repos/builtin/packages/icedtea/package.py
@@ -58,7 +58,7 @@ class Icedtea(AutotoolsPackage):
     depends_on('libxcomposite', when='~X', type='build')
     depends_on('libxau', when='~X', type='build')
     depends_on('libxdmcp', when='~X', type='build')
-    depends_on('gtkplus+X', when='~X', type='build')
+    depends_on('gtkplus', when='~X', type='build')
 
     depends_on('libx11', when='+X')
     depends_on('xproto', when='+X')
@@ -71,7 +71,7 @@ class Icedtea(AutotoolsPackage):
     depends_on('libxcomposite', when='+X')
     depends_on('libxau', when='+X')
     depends_on('libxdmcp', when='+X')
-    depends_on('gtkplus+X', when='+X')
+    depends_on('gtkplus', when='+X')
 
     depends_on('freetype@2:')
     depends_on('wget', type='build')

--- a/var/spack/repos/builtin/packages/py-pygtk/package.py
+++ b/var/spack/repos/builtin/packages/py-pygtk/package.py
@@ -38,7 +38,7 @@ class PyPygtk(AutotoolsPackage):
     depends_on('cairo')
     depends_on('glib')
     # for GTK 3.X use pygobject 3.X instead of pygtk
-    depends_on('gtkplus+X@2.24:2.99')
+    depends_on('gtkplus@2.24:2.99')
     depends_on('py-pygobject@2.28:2.99', type=('build', 'run'))
     depends_on('py-py2cairo', type=('build', 'run'))
 

--- a/var/spack/repos/builtin/packages/slurm/package.py
+++ b/var/spack/repos/builtin/packages/slurm/package.py
@@ -68,7 +68,7 @@ class Slurm(AutotoolsPackage):
     depends_on('readline')
     depends_on('zlib')
 
-    depends_on('gtkplus+X', when='+gtk')
+    depends_on('gtkplus', when='+gtk')
     depends_on('hdf5', when='+hdf5')
     depends_on('hwloc', when='+hwloc')
     depends_on('mariadb', when='+mariadb')

--- a/var/spack/repos/builtin/packages/wx/package.py
+++ b/var/spack/repos/builtin/packages/wx/package.py
@@ -47,7 +47,7 @@ class Wx(AutotoolsPackage):
     patch('math_include.patch', when='@3.0.1:3.0.2')
 
     depends_on('pkgconfig', type='build')
-    depends_on('gtkplus+X')
+    depends_on('gtkplus')
 
     @when('@:3.0.2')
     def build(self, spec, prefix):

